src/lib/transition_frontier/persistent_frontier/persistent_frontier.ml

```diff
@@ -52,21 +52,28 @@ let construct_staged_ledger_at_root
   in
   let mask = Ledger.of_database root_ledger in
   let%bind () =
-    Deferred.return
-      (List.fold transactions_with_protocol_state ~init:(Or_error.return ())
-         ~f:(fun acc (txn, protocol_state) ->
-           let open Or_error.Let_syntax in
-           let%bind () = acc in
-           let%map _ =
+    Deferred.Or_error.List.iter transactions_with_protocol_state
+      ~f:(fun (txn, protocol_state) ->
+        Deferred.return
```

Comment:
can this just be `let%map` ?

Reply:
The monad in the `iter` is `Deferred.Or_error.t`, but we're only operating over a `Or_error.t`, so I think we need this (or similar) as a monad conversion.